### ISSUE TITLE
I'm confused about how to make only the concept token embedding learnable

### ISSUE THREAD

**Issue by seilk** [callsys/GenPromp#3]
Thank you for the excellent paper and the implemented code. I have a point of confusion. In Figure 3 of the paper, only _v_r_  is colored in orange. 

Does this mean that among the embeddings for each word in "a photo of a <goldfish>", only the word embedding corresponding to <goldfish> is trainable?

However, I find the following part of your code confusing:

```python
# in datasets/base.py
def init_embeddings(self, text_encoder):
      token_embeds = text_encoder.get_input_embeddings().weight.data.clone()
      for token in self.cat2tokens:
          meta_token_id = self.tokenizer.encode(token['meta_token'], add_special_tokens=False)[0]
          concept_token_id = self.tokenizer.encode(token['concept_token'], add_special_tokens=False)[0]
          token_embeds[concept_token_id] = token_embeds[meta_token_id]
      text_encoder.get_input_embeddings().weight = torch.nn.Parameter(token_embeds)
      return text_encoder
```

This code sets token_embeds as trainable by making it a torch.nn.Parameter. 
However, contrary to what is shown in Figure 3, this seems to make the entire token_embeds trainable, not just the concept token vector corresponding to <goldfish>.

Could you please clarify my confusion? Thank you very much.

P.S. If my understanding is correct, I believe the following lines of code would be necessary to make only the concept token vector trainable:

```python
text_encoder.get_input_embeddings().weight.requires_grad = False
text_encoder.get_input_embeddings().weight[concept_token_id].requires_grad = True
```

**Comment by callsys** [comment]
In line 466 of `main.py`, we delete the grad for all tokens except the concept token, which makes it possible for only the concept token to back-propagate gradients during the training phase:
```
grads.data[index_grads_to_zero, :] = grads.data[index_grads_to_zero, :].fill_(0)
```

**Comment by seilk** [comment]
issue all successfully cleared
